util/git/client.go

```diff
@@ -151,15 +155,25 @@ func NewClientExt(rawRepoURL string, root string, creds Creds, insecure bool, en
 //   a client with those certificates in the list of root CAs used to verify
 //   the server's certificate.
 // - Otherwise (and on non-fatal errors), a default HTTP client is returned.
-func GetRepoHTTPClient(repoURL string, insecure bool, creds Creds) *http.Client {
+func GetRepoHTTPClient(repoURL string, insecure bool, creds Creds, proxyURL string) *http.Client {
 	// Default HTTP client
 	var customHTTPClient = &http.Client{
 		// 15 second timeout
 		Timeout: 15 * time.Second,
-		// don't follow redirect
-		CheckRedirect: func(req *http.Request, via []*http.Request) error {
-			return http.ErrUseLastResponse
-		},
+	}
+
+	proxyFunc := proxy.GetCallback(proxyURL)
+
+	if IsHTTPURL(repoURL) {
+		customHTTPClient.Transport = &http.Transport{
+			Proxy: proxyFunc,
+		}
+		return customHTTPClient
+	}
```

Comment:
The function `GetRepoHTTPClient` created the custom client for `https` protocols. By differentiating `http` here, we return the client with just the proxy and further configurations(insecure,certs,etc) are added for only `https`.

Reply:
IMHO, we can just use the same transport for HTTP and HTTPS protocols. There's no need to differentiate. All TLS related settings in the transport would only be used when connecting to a HTTPS backend (I think the `net/http` library decides that on the protocol designator from the URL), and is ignored otherwise.

Also, the returned client above is _incomplete_ in that it doesn't set the redirect callback to prevent following redirects.

Reply:
Thanks! Updated to use the same transport